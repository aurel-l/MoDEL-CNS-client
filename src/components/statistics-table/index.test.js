import React from 'react';
import { shallow } from 'enzyme';

import StatisticsTable from '.';

describe('<StatisticsTable />', () => {
  it('should render empty table', () => {
    expect(shallow(<StatisticsTable />)).toMatchSnapshot();
  });

  it('should render table with data', () => {
    expect(
      shallow(
        <StatisticsTable
          y={{
            rgyr: { average: Math.PI, stddev: 1 },
            'other-analysis': { average: 2, stddev: 2 },
          }}
        />,
      ),
    ).toMatchSnapshot();
  });
});
