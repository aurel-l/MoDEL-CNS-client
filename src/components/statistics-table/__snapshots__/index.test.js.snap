// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatisticsTable /> should render empty table 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)>
        name
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        mean
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        standard deviation
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody) />
</WithStyles(Table)>
`;

exports[`<StatisticsTable /> should render table with data 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)>
        name
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        mean
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        standard deviation
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      key="rgyr"
    >
      <WithStyles(TableCell)>
        Rgyr
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        3.1416
        nm
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        1.0000
        nm
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;
